src/supertux/game_session.cpp

```diff
@@ -282,6 +298,23 @@ GameSession::on_escape_press(bool force_quick_respawn)
   }
 }
 
+Vector
+GameSession::get_fade_point() const
+{
+  // Get first player that is alive
+  // Should work for single player without problems,
+  // but for multiplayer a proper handling needs to be done
+  for (const auto& player : m_currentsector->get_players())
+  {
+    if (!player->is_dead() && !player->is_dying())
+    {
+      return player->get_bbox().get_middle();
+    }
+  }
+
+  return Vector(0.0f, 0.f);
+}
+
```

Comment:
Perhaps the fade could be done where the door is, instead of where the first alive player is?

For the script, a parameter as to where to do the fade at could work.

Reply:
This function is only used when the point cannot be determined automatically. If there is a door, that triggers the respawn, its coordinates are used (see door passing its center to the respawn_with_fade function).
This functions is mainly for the scripting part where the trigger cannot be determined (or at least I did not figure out how to get it in case of the "Level.*" commands or if invoked from console.

Reply:
For the scripting function, screen-relative coordinates could be required.